src/pages/index.js

```diff
@@ -100,36 +146,95 @@ function fillProfileForm() {
     descInput.value = info.description;
 }
 
-function updateInfo(event, data) {
-    userInfo.setUserInfo({
+async function updateInfo(event, data) {
+    const newInfo = {
         name: data["name-input"],
         description: data["description-input"],
+    };
+    userInfo.setUserInfo(newInfo);
+    api.editProfile({
+        name: newInfo.name,
+        about: newInfo.description
     });
 }
 
+function openAvatarModal() {
+    avatarPopup.open();
+    formValidators["avatar-form"].toggleButtonState();
+}
+
+async function updateAvatar(event, data) {
+    const link = data["url-input"];
+    api.updateAvatar(link)
+        .then((res) => {
+            document.querySelector(".profile__image")
+                .setAttribute("src", res.avatar);
+        });
+    
+}
+
 // #endregion 
 
-// #region New place modal & card creation methods
+// #region New place modal & card methods
 
 function openPlaceModal() {
     formValidators["place-form"].toggleButtonState();
     placePopup.open();
 }
 
-function addCard(event, data) {
+function openConfirmModal(event, data) {
+    deletePopup.open();
+    deleteButton.removeEventListener("click", deleteHandler);
+    deleteHandler = () => {
+        deleteCard(event.target.closest(".card"), data);
+    };
+    deleteButton.addEventListener("click", deleteHandler);
+}
+
+async function addCard(event, data) {
     const cardData = {
         name: data["title-input"],
         link: data["url-input"],
     };
-    cardsSection.addItem(getCardElement(cardData));
+
+    api.addCard(cardData)
+        .then((res) => {
+            cardData._id = res._id;
+            cardsSection.addItem(getCardElement(cardData));
+        });
+
     event.target.reset();
 }
 
+function deleteCard(cardElement, data) {
+    cardElement.remove();
+    api.deleteCard(data._id)
+    deletePopup.close();
+}
+
+function likeCard(event, data) {
+    const button = event.target;
+    button.classList.toggle("card__like-button_active");
+    if (data.isLiked) {
+        api.unlikeCard(data._id);
+        data.isLiked = false;
+    } else {
+        api.likeCard(data._id);
+        data.isLiked = true;
+    }
+}
+
 function getCardElement(data) {
     const cardSelector = "card";
     const card = new Card(data, cardSelector,
-        () => { imageModal.open(data) });
+        () => { imageModal.open(data) },
+        (event) => { openConfirmModal(event, data) },
+        (event) => { likeCard(event, data) });
     const cardElement = card.generateCard();
+    if (data.isLiked) {
+        const likeButton = cardElement.querySelector(".card__like-button");
+        likeButton.classList.toggle("card__like-button_active");
```

Comment:
```js

    if (data.isLiked) {
        const likeButton = cardElement.querySelector(".card__like-button");
        likeButton.classList.toggle("card__like-button_active");
    }
```

This should be done only inside `Card`


* Now only class `Card` knows every card selector, so you need to encapsulate the likes logics inside its methods and call them in `index.js`

Reply:
You haven't corrected my remark https://github.com/YuRo711/se_project_aroundtheus/pull/1#discussion_r1459560822